apps/entity-service/src/entities/processors/site.processor.ts

```diff
@@ -1,24 +1,161 @@
+import { DocumentBuilder } from "@terramatch-microservices/common/util";
+import { Aggregate, aggregateColumns, EntityProcessor, PaginatedResult } from "./entity-processor";
+import {
+  Demographic,
+  DemographicEntry,
+  Media,
+  ProjectUser,
+  Seeding,
+  Site,
+  SitePolygon,
+  SiteReport,
+  TreeSpecies
+} from "@terramatch-microservices/database/entities";
+import { AdditionalSiteFullProps, SiteFullDto, SiteLightDto, SiteMedia } from "../dto/site.dto";
+import { EntityQueryDto } from "../dto/entity-query.dto";
+import { BadRequestException } from "@nestjs/common";
+import { FrameworkKey } from "@terramatch-microservices/database/constants/framework";
+import { Op } from "sequelize";
+import { sumBy } from "lodash";
 
 export class SiteProcessor extends EntityProcessor<Site, SiteLightDto, SiteFullDto> {
   readonly LIGHT_DTO = SiteLightDto;
   readonly FULL_DTO = SiteFullDto;
 
-  async findOne(): Promise<Site> {
-    throw new Error("Method not implemented.");
+  async findOne(uuid: string) {
+    return await Site.findOne({
+      where: { uuid },
+      include: [
+        { association: "framework" },
+        { association: "project", attributes: ["name"], include: [{ association: "organisation" }] }
+      ]
+    });
   }
 
-  async findMany(): Promise<PaginatedResult<Site>> {
-    throw new Error("Method not implemented.");
+  async findMany(query: EntityQueryDto, userId?: number, permissions?: string[]): Promise<PaginatedResult<Site>> {
+    const builder = await this.entitiesService.buildQuery(Site, query, [
+      { association: "project", attributes: ["name"] },
+      { association: "framework" }
+    ]);
+
+    if (query.sort != null) {
+      if (["name"].includes(query.sort.field)) {
+        builder.order([query.sort.field, query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field === "projectName") {
+        builder.order(["project", "name", query.sort.direction ?? "ASC"]);
+      } else if (query.sort.field !== "id") {
+        throw new BadRequestException(`Invalid sort field: ${query.sort.field}`);
+      }
+    }
+
+    const frameworkPermissions = permissions
+      ?.filter(name => name.startsWith("framework-"))
+      .map(name => name.substring("framework-".length) as FrameworkKey);
+    if (frameworkPermissions?.length > 0) {
+      builder.where({ frameworkKey: { [Op.in]: frameworkPermissions } });
+    } else if (permissions?.includes("manage-own")) {
+      builder.where({
+        projectId: { [Op.in]: ProjectUser.userProjectsSubquery(userId) }
+      });
+    } else if (permissions?.includes("projects-manage")) {
+      builder.where({
+        projectId: { [Op.in]: ProjectUser.projectsManageSubquery(userId) }
+      });
+    }
+
+    for (const term of ["name", "projectName"]) {
+      if (query[term] != null) builder.where({ [term]: query[term] });
+    }
+    if (query.search != null) {
+      builder.where({ name: { [Op.like]: `%${query.search}%` } });
```

Comment:
The existing implementation for sites (see `Site.php` `search` method) matches the name search on sites against site name, project name or org name. I think org name is a bit odd, especially since we don't do that on projects, but I would at least include project name here. Or check with product. Even that feels a bit weird to me.